File: src/main/java/com/msg/gauth/domain/auth/services/LogoutService.java
package com.msg.gauth.domain.auth.services;

import com.msg.gauth.domain.auth.RefreshToken;
import com.msg.gauth.domain.auth.repository.RefreshTokenRepository;
import com.msg.gauth.domain.user.User;
import com.msg.gauth.domain.user.utils.UserUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LogoutService {
    private final UserUtil userUtil;
    private final RefreshTokenRepository refreshTokenRepository;

    void execute() {
        User currentUser = userUtil.fetchCurrentUser();
        refreshTokenRepository.findById(currentUser.getId())
                .ifPresent(refreshTokenRepository::delete);
    }
}

File: src/main/kotlin/com/msg/gauth/domain/auth/services/LogoutService.kt
package com.msg.gauth.domain.auth.services

import com.msg.gauth.domain.auth.repository.RefreshTokenRepository
import com.msg.gauth.domain.user.utils.UserUtil
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class LogoutService(
    private val userUtil: UserUtil,
    private val refreshTokenRepository: RefreshTokenRepository
) {
    @Transactional(rollbackFor = [Exception::class])
    fun execute() {
        val currentUser = userUtil.fetchCurrentUser()
        refreshTokenRepository.findByUserId(currentUser.id)?.let {
            refreshTokenRepository.delete(it)
        }
    }
}